hackathon_site/hardware/serializers.py

```diff
@@ -0,0 +1,41 @@
+from rest_framework import serializers
+from hardware.models import Hardware, Category, Order
+
+
+class HardwareSerializer(serializers.ModelSerializer):
+    quantity_remaining = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Hardware
+        fields = (
+            "id",
+            "name",
+            "model_number",
+            "manufacturer",
+            "datasheet",
+            "quantity_available",
+            "notes",
+            "max_per_team",
+            "picture",
+            "categories",
+            "quantity_remaining",
+        )
+
+    def get_quantity_remaining(self, obj):
```

Comment:
You can make this a static method (I also like to type annotate for code hints): 

```
@staticmethod
def get_quantity_remaining(obj: Hardware):
    return (
        obj.quantity_available
        - Order.objects.filter(items__hardware__id=obj.id).count()
    )
```

Reply:
Is this within the class or no?

Reply:
Yup, exactly where it is, just with the decorator on top. Static methods don't need the "self" parameter.